fees/manager.go

```diff
@@ -78,19 +85,22 @@ func (f *Manager) LastConsumed(d Dimension) uint64 {
 }
 
 func (f *Manager) lastConsumed(d Dimension) uint64 {
-	start := dimensionStateLen*d + consts.Uint64Len + window.WindowSliceSize
+	start := consts.IntLen + dimensionStateLen*d + consts.Uint64Len + window.WindowSliceSize
 	return binary.BigEndian.Uint64(f.raw[start : start+consts.Uint64Len])
 }
 
-func (f *Manager) ComputeNext(lastTime int64, currTime int64, r Rules) (*Manager, error) {
+func (f *Manager) ComputeNext(currTime int64, r Rules) (*Manager, error) {
 	f.l.RLock()
 	defer f.l.RUnlock()
 
 	targetUnits := r.GetWindowTargetUnits()
 	unitPriceChangeDenom := r.GetUnitPriceChangeDenominator()
 	minUnitPrice := r.GetMinUnitPrice()
-	since := int((currTime - lastTime) / consts.MillisecondsPerSecond)
-	bytes := make([]byte, dimensionStateLen*FeeDimensions)
+	lastTimeSeconds := int64(binary.BigEndian.Uint64(f.raw[0:consts.Int64Len]))
+	currTimeSeconds := int64(currTime / consts.MillisecondsPerSecond)
+	since := uint64(currTimeSeconds - lastTimeSeconds)
+	bytes := make([]byte, consts.Int64Len+dimensionStateLen*FeeDimensions)
+	binary.BigEndian.PutUint64(bytes[0:consts.Int64Len], uint64(currTimeSeconds))
```

Comment:
By storing the timestamp in seconds, as soon as we move 1.9 to 2.1 it will be treated as a 1 second interval elapsing. Previously, this would have been treated as 200ms rounding down to no difference.